uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -1540,6 +1540,42 @@ sub my_trim {
 	return $str;
 }
 
+=pod
+
+=head3 fetch_minc_header_info($minc, $field, $awk, $keep_semicolon)
+
+Function that fetches header information in MINC file.
+
+INPUTS:
+  - $minc : MINC file
+  - $field: field name to look for in MINC header (or 'all' to grep all headers)
+  - $awk  : awk info to check if argument inserted in MINC header
+  - $keep_semicolon: if set, keep ";" at the end of extracted value
+
+RETURNS: value of the field found in the MINC header
+
+=cut
+
+sub fetch_header_info {
+    my ($minc, $field, $awk, $keep_semicolon) = @_;
+    my $value;
+    if ($field eq 'all') {
+        # run mincheader and return all the content of the command
+        $value = `mincheader -data "$minc"`;
+    }
+    else {
+        # fetch a particular header value, remove extra spaces and optionally
+        # the semicolon
```

Comment:
Why is the `-data` switch used in the call to `mincheader` above but not in the one below? I am a bit surprised also that there is not an option in `mincheader` to retrieve a specific header (there is one in `dcmdump`). Oh well...

Reply:
From the mincheader manual:
```If the input file  is compressed, then its header is decompressed and only the  header  is  printed.  If  the -data option is given, then the whole file is decompressed and all data (except the image) is dumped.```

For fun I tried running mincheader on a MINC file with the option `-data` and without and compared the outputs. They are exactly the same. I think there is a difference in using the `-data` option in old `mnc.gz` files?

The reason it is in the first `if` statement is that it comes from the File.pm library which is old and probably needed to use that `-data` option on `mnc.gz` files. The `mincheader` in the second `if` statement was created way later and I did not know that a `-data` option existed ;).

Just to be on the safe side, I will add the option `-data` to the second statement.

Reply:
done